src/edu/ucsb/nceas/metacat/admin/upgrade/HashStoreUpgrader.java

```diff
@@ -66,16 +66,28 @@ public class HashStoreUpgrader implements UpgradeUtilityInterface {
     private File backupDir;
     private static int timeout = TIME_OUT_DAYS;
     private int maxSetSize = MAX_SET_SIZE;
-    private static int nThreads = 1;
+    protected static int nThreads;
 
     static {
-        // use a shared executor service with nThreads == one less than available processors or one
-        int availableProcessors = Runtime.getRuntime().availableProcessors();
-        nThreads = availableProcessors;
-        nThreads--;
-        nThreads = Math.max(1, nThreads);
+        // use a shared executor service with nThreads == one less than available processors (or one
+        // if there's only 1 processor), and limited by the max number of database connections.
+        int availDbConn;
+        try {
+            // Limit to DB conn pool size minus 5 for other processes
+            availDbConn =
+                Integer.parseInt(PropertyService.getProperty("database.maximumConnections")) - 5;
```

Comment:
Since `availDbConn = Constance - 5`, it can a zero or negative number.  Do you think we should add a line after line 78:
```
availDbConn =  Math.max(1, availDbConn);
```

Reply:
good catch - fixed